[WIP] Makefile: fix bin/docgen in cross compilation

otan opened the pull request:
`bin/docgen` imports builtins, which imports `geoproj`, which imports a
C library. Unfortunately, we try to install docgen in the Makefile but
will error as it is incompatible with the host system during cross
compilation. Hence, we special case some hacky fixes around to cover
these cases.

Also removed unnecessary imports in the zcgo_flags.go for GEOS and PROJ
in the CPP flags.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/49897)
<!-- Reviewable:end -->

otan commented:
urgh, passed locally and failed on ci....

otan committed:
Makefile: fix bin/docgen in cross compilation

`bin/docgen` imports builtins, which imports `geoproj`, which imports a
C library. Unfortunately, we try to install docgen in the Makefile but
will error as it is incompatible with the host system during cross
compilation. Hence, we special case some hacky fixes around to cover
these cases.

Also removed unnecessary imports in the zcgo_flags.go for GEOS and PROJ
in the CPP flags.

Release note: None

otan commented:
have a better solution than this i think, will try in the morning.
